### DISCUSSION TITLE
Is it possible to use ffmpeg filters on the video stream, not just movie events?

### DISCUSSION THREAD

**Discussion by bottlehall** [Motion-Project/motion#1385] [Q&A]
0. Reviewed guide and contributing documents? (Yes/No): Yes
1. version [x.y.z, hash, other]: 4.3.2
2. installed as a package or compiled from sources [deb, rpm, git, other]: Debian via apt
3. standalone or part of third party [motion, MotionEyeOS, other]: standalone
4. video stream source [V4L (card or USB), net cam (mjpeg, rtsp, other), mmal]:  v4l
5. hardware [x86, ARM, other]: Arm
6. operating system [16.04, Stretch, etc, FreeBSD, other]: Ubuntu 20.04

I would like to be able to do keystone-correction on the streamed image. I can achieve this using ffmpeg to apply a filter and stream the result, but RTMP/HLS lags/hangs in the browser unacceptably. Using motion to stream would seem to be a much better option, but I can't find a way of applying the ffmpeg filter other than during motion events using `movie_extpipe` or `movie_extpipe_use`. Is there a way of applying filters to the real-time stream?

**Comment by Mr-Dave** [comment]
You could try the second [static file](https://motion-project.github.io/motion_config.html#basic_setup_static) process indicated in the instructions except use whatever input you'd like to correct.
